src/common/appdb/models/saved_connection.ts

```diff
@@ -217,13 +196,38 @@ export class SavedConnection extends DbConnectionBase {
       looseMatching: false
     }
   })
-  sshPassword
+  sshPassword: Nullable<string> = null
+
+
+
+  _sshMode: string = "agent"
+
+  @Column({name: "sshMode", type: "varchar", length: "8", nullable: false, default: "agent"})
+  set sshMode(value: string) {
+    this._sshMode = value
+    // TODO: (geovannimp) This code dont make sense, have to be refactored
```

Comment:
Yes, I think this should just be:

`if(this._sshMode !== 'userpass')`

IE - clear the SSH password if not authenticating with password

Reply:
Ok, but with this the `else if` don't work.

If `_sshMode` is not different from `"userpass"`, that means that `_sshMode` is `"userpass"` and cant be `"keyfile"`.

Typescript give me the following erro: 

```
This condition will always return 'true' since the types '"userpass"' and '"keyfile"' have no overlap.ts(2367)
```

Btw, I changed the if/else to only else. Only confirm to me that I catch this right.